### PR TITLE
Fix/adf 515/criteria search fixes

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/ADF-515

Changes made:

- [x] Refactored metadata values storage structuring ( instead of labels, which may be subject to change due to i18n and renaming, kept to immutable `uri` ) 
- [x] Logger improvements and fixes
- [x] Minor cleaning of comments and old styled formatting

### CODE REVIEW

File: src/ElasticSearch.php [hunk 1]
```diff
@@ -121,26 +111,26 @@ public function query($queryString, $type, $start = 0, $count = 10, $order = '_i
...
             switch ($exception->getCode()) {
                 case 400:
                     $json = json_decode($exception->getMessage(), true);
                     $message = __(
                         'There is an error in your search query, system returned: %s',
-                        $json['error']['reason']
+                        $json['error']['reason'] ?? ''
                     );
-                    $this->getLogger()->error($message, [$exception->getMessage()]);
+                    $this->getLogger()->error(sprintf('Elasticsearch: %s %s', $message, $exception->getMessage()));
```
Comment (gabrielfs7) [inline src/ElasticSearch.php:129]: Do you think it helps to put in the logs the exception FQN too? Sometimes the message is too generic and we do not know from where it comes :)

File: src/QueryBuilder.php [hunk 2]
```diff
@@ -184,10 +180,8 @@ private function buildCustomConditions(QueryBlock $queryBlock): string
     {
         $conditions = [];
 
-        $field_slug = tao_helpers_Slug::create($queryBlock->getField());
-
         foreach (self::CUSTOM_FIELDS as $customField) {
-            $conditions[] = sprintf('%s_%s:"%s"', $customField, $field_slug, $queryBlock->getTerm());
+            $conditions[] = sprintf('%s_%s:"%s"', $customField, $queryBlock->getField(), $queryBlock->getTerm());
```
Comment (gabrielfs7) [inline src/QueryBuilder.php:184]: I imagine yes, but just to be 100% sure, will the `$queryBlock->getField()` be always URI encoded from tao?

**Review by gabrielfs7 — APPROVED** [review]
Good job! 🚀 

Some minor suggestions :) 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by shpran — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Pull request merged**